shared/src/bundles/user-details/types/user-details-create-dto.type.ts

```diff
@@ -0,0 +1,44 @@
+import { type ValueOf } from '~/types/value-of.type.js';
+
+import {
+    type CountryList,
+    type EmploymentType,
+    type EnglishLevel,
+    type JobTitle,
+    type NotConsidered,
+    type PreferredLanguages,
+} from '../user-details.js';
+
+type UserDetailsCreateDto = {
+    userId: string;
+
+    profileName?: string;
+
+    salaryExpectation?: number;
+    hiredSalary?: number;
+
+    jobTitle?: ValueOf<typeof JobTitle>;
+    location?: ValueOf<typeof CountryList>;
+
+    experienceYears?: number;
+    employmentType?: ValueOf<typeof EmploymentType>[];
+
+    description?: string;
+
+    englishLevel?: ValueOf<typeof EnglishLevel>;
+    notConsidered?: ValueOf<typeof NotConsidered>[];
+    preferredLanguages?: ValueOf<typeof PreferredLanguages>[];
+
+    projectLinks?: string[];
+    photoId?: string;
+    fullName: string;
```

Comment:
We can make it optional (As improvement)
Example
```js
            const newUserDetails = await this.userDetailsRepository.create({
                fullName: fullName ?? '',
                ...payload,
            });
```

Reply:
added